source/kraken/apply_disruption.cpp

```diff
@@ -583,7 +586,11 @@ struct delete_impacts_visitor : public apply_impacts_visitor {
         // it with an empty vector.
         decltype(mvj->impacted_by) impacted_by_moved;
         boost::swap(impacted_by_moved, mvj->impacted_by);
-        disruptions_collection.insert(std::begin(impacted_by_moved), std::end(impacted_by_moved));
+        
+        // note: insert(it_begin, it_end) crashed...
+        for(const auto& i: impacted_by_moved) {
```

Comment:
With a correct "complete" comparator it still crashes?
Problem of references or something like that?
Would be interesting to investigate a little on that even after merging, this can be a real problem (memcheck errors?)